tests/Immediate.Handlers.FunctionalTests/Behavior/Constraints/Tests.BehaviorShouldConstrain_A.cs

```diff
@@ -0,0 +1,41 @@
+using Immediate.Handlers.Shared;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Immediate.Handlers.FunctionalTests.Behavior.Constraints;
+
+[Handler]
+[Behaviors(typeof(BehaviorA<,>), typeof(BehaviorB<,>), typeof(BehaviorC<,>), typeof(BehaviorD<,>))]
+public static partial class BehaviorShouldConstrainA
+{
+	public sealed record Query(int Input) : A;
+
+	private static Task<int> Handle(
+		Query query,
+		CancellationToken _)
+	{
+		return Task.FromResult(query.Input + 1);
+	}
+}
+
+public sealed partial class Tests
+{
+	[Fact]
+	public async Task BehaviorShouldConstrain_A()
+	{
+		IServiceCollection services = new ServiceCollection();
+		services = ConfigureBehaviors(services);
+		services = BehaviorShouldConstrainA.AddHandlers(services);
+		var serviceProvider = services.BuildServiceProvider();
+
+		var handler = ActivatorUtilities.CreateInstance<BehaviorShouldConstrainA.Handler>(serviceProvider);
+		_ = await handler.HandleAsync(new(1));
+
+		var behaviorWalker = serviceProvider.GetRequiredService<BehaviorWalker>();
+
+		Assert.Contains("BehaviorA", behaviorWalker.BehaviorsRan);
+		Assert.DoesNotContain("BehaviorB", behaviorWalker.BehaviorsRan);
+		Assert.DoesNotContain("BehaviorC", behaviorWalker.BehaviorsRan);
+		Assert.DoesNotContain("BehaviorD", behaviorWalker.BehaviorsRan);
```

Comment:
Can be simplified as:
```cs
Assert.Equal(["BehaviorA"], behaviorWalker.BehaviorsRan);
```

(may require `BehaviorsRan` to be a `List<string>` instead.